File: app/src/main/java/com/example/james/rms/Core/ServePath/ProductServerPath.java
package com.example.james.rms.Core.ServePath;

import com.example.james.rms.NetWork.ServeProfile;

/**
 * Created by jamie on 2017/4/17.
 */

public class ProductServerPath {

public static String serve_findAll(){
    String serve = ServeProfile.getServe();
    String path = ServeProfile.getProduct_findAll();
    String serve_path = serve+path;
    return serve_path;
}

    public static String serve_findByPartyId(){
        String serve = ServeProfile.getServe();
        String path = ServeProfile.getProduct_findByPartyId();
        String serve_path = serve+path;
        return serve_path;
    }

    public static String serve_insertProduct(){
        String serve = ServeProfile.getServe();
        String path = ServeProfile.getProduct_insert();
        String serve_path = serve+path;
        return serve_path;
    }
    public static String serve_updateWeightIdNullByWeightIdAndPartyId(){
        String serve = ServeProfile.getServe();
        String path = ServeProfile.getProduct_updateWeightIdNullByWeightIdAndPartyId();
        String serve_path = serve+path;
        return serve_path;
    }
    public static String serve_updateQuantityIdNullByQuantityIdAndPartyId(){
        String serve = ServeProfile.getServe();
        String path = ServeProfile.getProduct_updateQuantityIdNullByQuantityIdAndPartyId();
        String serve_path = serve+path;
        return serve_path;
    }

    public static String serve_findByProductId(){
        String serve = ServeProfile.getServe();
        String path = ServeProfile.getProduct_findByProductId();
        String serve_path = serve+path;
        return serve_path;
    }

    public static String serve_updateQuantityIdAndWeightIdNullByProductId(){
        String serve = ServeProfile.getServe();
        String path = ServeProfile.getProduct_updateQuantityIdAndWeightIdNullByProductId();
        String serve_path = serve+path;
        return serve_path;
    }

    public static String serve_deleteByProductId(){
        String serve = ServeProfile.getServe();
        String path = ServeProfile.getProduct_deleteByProductId();
        String serve_path = serve+path;
        return serve_path;
    }

    public static String serve_delete(){
        String serve = ServeProfile.getServe();
        String path = ServeProfile.getProduct_delete();
        String serve_path = serve+path;
        return serve_path;
    }

}

File: app/src/main/java/com/example/james/rms/core/serve_path/ProductServerPath.kt
package com.example.james.rms.core.serve_path

import com.example.james.rms.network.ServeProfile

/**
 * Created by jamie on 2017/4/17.
 */

object ProductServerPath {
    val serve: String
        get() = ServeProfile.serve

    fun serve_findAll(): String {
        val path = ServeProfile.product_findAll
        return serve + path
    }

    fun serve_findByPartyId(): String {
        val path = ServeProfile.product_findByPartyId
        return serve + path
    }

    fun serve_insertProduct(): String {
        val path = ServeProfile.product_save
        return serve + path
    }

    fun serve_updateWeightIdNullByWeightIdAndPartyId(): String {
        val path = ServeProfile.product_updateWeightIdNullByWeightIdAndPartyId
        return serve + path
    }

    fun serve_updateQuantityIdNullByQuantityIdAndPartyId(): String {
        val path = ServeProfile.product_updateQuantityIdNullByQuantityIdAndPartyId
        return serve + path
    }

    fun serve_findByProductId(): String {
        val path = ServeProfile.product_findByProductId
        return serve + path
    }

    fun serve_updateQuantityIdAndWeightIdNullByProductId(): String {
        val path = ServeProfile.product_updateQuantityIdAndWeightIdNullByProductId
        return serve + path
    }

    fun serve_deleteByProductId(): String {
        val path = ServeProfile.product_deleteByProductId
        return serve + path
    }

    fun serve_delete(): String {
        val path = ServeProfile.product_delete
        return serve + path
    }

}
